src/c_api.cpp

```diff
@@ -175,8 +175,11 @@ class Booster {
     } else {
       is_raw_score = false;
     }
+    auto param = ConfigBase::Str2Map(parameter);
```

Comment:
I'm really concerned about this. Do you know how long Str2Map() takes? We'll be doing this for each instance we want to classify. IMO it would be worth benchmarking the after/before to see how this 'hurts' performance when only a few trees are being used (with early stopping disabled)